pkg/config/config.go

```diff
@@ -248,8 +269,35 @@ func CustomHTTPErrorHandler(err error, c echo.Context) {
 	}
 }
 
-func ConfigureEcho() *echo.Echo {
-	e := echo.New()
+func metricsMiddlewareSkipper(ctx echo.Context) bool {
+	path := ctx.Request().URL.Path
+	switch {
+	case path == "/ping" || path == "/ping/":
+		return true
+	case path == "/metrics" || path == "/metrics/":
+		return true
+	}
+	pathItems := handler_utils.NewPathWithString(path).RemovePrefixes()
+	if pathItems.StartWithResources(
+		[]string{"ping"},
+		[]string{"repositories", "validate"},
+		[]string{"repository_parameters", "validate"},
+	) {
+		return true
+	}
+	return false
+}
+
+func createMetricsMiddleware(metrics *instrumentation.Metrics) echo.MiddlewareFunc {
+	return instrumentation.MetricsMiddlewareWithConfig(
+		&instrumentation.MetricsConfig{
+			Skipper: metricsMiddlewareSkipper,
+			Metrics: metrics,
+		})
+}
+
+func configureEchoCommon(e *echo.Echo, metrics *instrumentation.Metrics) *echo.Echo {
```

Comment:
Since `configureEchoCommon` takes metrics but does not use it, I suggest changing the echo configuration functions to [these](https://gist.github.com/rverdile/150de14374ee08d0d182daa16f8e5033). 

With this there's only two functions: one called when creating the API server, and the other called when creating the metrics server. Hopefully that makes sense.